internal/ascpexecauth/postgres.go

```
@@ -3,50 +3,148 @@ package ascpexecauth
+	if !validInput(input) || input.ApprovalID == "" || !reservationExpiresAfter(input, now) ||
+		input.Reservation.ExpiresAt.After(now.Add(ReservationTTL)) {
 		return Authorization{}, ErrInvalidInput
 	}
+	if err := validateReservation(input.Reservation); err != nil {
+		return Authorization{}, fmt.Errorf("%w: %v", ErrInvalidInput, err)
+	}
 	tx, err := s.db.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSerializable})
 	if err != nil {
 		return Authorization{}, fmt.Errorf("begin execution authorization transaction: %w", err)
 	}
 	defer tx.Rollback()
```

golangci-lint (2.12.2)

[error] Error return value of `tx.Rollback` is not checked

(errcheck)